nexus/types/src/deployment/planning_report.rs

```diff
@@ -501,11 +500,144 @@ impl PlanningMupdateOverrideStepReport {
 #[serde(rename_all = "snake_case")]
 #[serde(tag = "type", content = "value")]
 #[cfg_attr(test, derive(test_strategy::Arbitrary))]
-// TODO-K: Separate into enums for each component as suggested in
-// https://github.com/oxidecomputer/omicron/pull/9001#discussion_r2372863166
-// and including more detailed information as suggested in
-// https://github.com/oxidecomputer/omicron/pull/9001#discussion_r2372842378
 pub enum FailedMgsUpdateReason {
+    /// There was a failed Host OS update
+    #[error("failed Host OS update: {0:?}")]
+    HostOs(FailedHostOsUpdateReason),
```

Comment:
All of these should be using error sources and not embedding debug strings; e.g.,


```suggestion
    #[error("failed Host OS update")]
    HostOs(#[from] FailedHostOsUpdateReason),
```

(We also have https://github.com/oxidecomputer/omicron/blob/main/docs/error-types-and-logging.adoc if you want much more detail on this.)

Should the error messages also be "failed _to plan_"? As written it sounds like we failed to perform an update.